### PR TITLE
Graphed validation for the DETR-line campaign families

### PR DESCRIPTION
**Pull request by EHxuban11**
## Summary

- Opt `rfdetr`, `dfine`, `deimv2` and `ec` into `SUPPORTS_CUDA_GRAPH` so training-time validation can capture their eval forwards (RF100-VL perf-coverage PRD, item P1.2). Validation is launch-bound for the small variants those campaigns run, so this is the biggest remaining epoch-time lever for the DETR MUST families.
- Add per-family parity gates in `tests/unit/test_cuda_graph_detr_families.py` (the `test_validator_cuda_graph.py` mould): opt-in surface + clean CPU fallback, CUDA capture/replay bit-parity vs eager at the model level, and at the validator level — the exact wiring training-time validation uses (up-front capture, replay-only loop policy, graph scope).
- Verify YOLOv9-E2E's inherited flag (PRD item P0.3): the e2e dual-assignment head's eval forward is included in every gate as its own case and captures/replays bit-identically, so the inheritance from `LibreYOLO9` is proven safe and no `SUPPORTS_CUDA_GRAPH = False` override is needed.

Why these forwards are graph-safe: D-FINE/DEIMv2/EC share the FDR decoder line — pure tensor work at a fixed `eval_spatial_size`, lazily built anchors settle during capture warmup, top-k has a static output shape. RF-DETR's deformable attention uses atomics only in its backward, never in the inference forward.

Not included, deliberately: training capture specs for dfine/deimv2/ec (PRD P1.3). Their training forward passes `targets` into the network to build contrastive-denoising query groups whose count varies with per-image GT, which violates the capture contract (target-free forward, static shapes). Documented as infeasible without restructuring denoising.

## Test plan

- [x] `pytest tests/unit/test_cuda_graph_detr_families.py` — 20/20 pass on CUDA hardware (RTX 5070 Ti, fp32): 10 CPU (opt-in + fallback), 5 model-level capture/replay bit-parity, 5 validator-level wiring bit-parity.
- [ ] CI (CPU-only) runs the 10 CPU tests; CUDA gates are skipped there by design.

<details><summary><h3>Greptile Summary</h3></summary>

The PR opts four DETR-line model families into CUDA-graph validation and adds CPU fallback, model replay, and validator-wiring parity coverage. The RF-DETR test isolation is improved, but its constructor still introduces an external weight-download dependency into the unit suite.
- Enables `SUPPORTS_CUDA_GRAPH` for RF-DETR, D-FINE, DEIMv2, and EC.
- Adds shared parity gates for those families and YOLOv9-E2E.
- Defers RF-DETR imports until its parameterized cases are built.
</details>

<details><summary><h3>Confidence Score: 4/5</h3></summary>

The PR is not yet safe to merge because the new RF-DETR unit cases can fail in offline environments despite the optional-package guard.

The package-level skip fixes collection when Transformers is absent, but when it is installed the size-n RF-DETR constructor still reaches `AutoModel.from_pretrained`, making the unit suite depend on remote Hugging Face availability.

**Files Needing Attention:** tests/unit/test_cuda_graph_detr_families.py
</details>

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| libreyolo/models/deimv2/model.py | Enables CUDA-graph validation for DEIMv2 with parity coverage in the new tests. |
| libreyolo/models/dfine/model.py | Enables CUDA-graph validation for D-FINE with parity coverage in the new tests. |
| libreyolo/models/ec/model.py | Enables CUDA-graph validation for EC with parity coverage in the new tests. |
| libreyolo/models/rfdetr/model.py | Enables CUDA-graph validation for RF-DETR. |
| tests/unit/test_cuda_graph_detr_families.py | Adds comprehensive parity coverage, but RF-DETR construction can download DINOv2 weights and fail in offline unit-test environments. |

</details>

<a href="https://app.greptile.com/api/ide/cursor?prompt=%23%23%23%20Issue%201%0Atests%2Funit%2Ftest_cuda_graph_detr_families.py%3A46%0A**RF-DETR%20unit%20tests%20require%20network%20access**%0A%0AIf%20%60transformers%60%20is%20installed%20but%20the%20DINOv2%20weights%20are%20not%20cached%20and%20Hugging%20Face%20is%20unavailable%2C%20this%20random-init%20construction%20still%20reaches%20%60AutoModel.from_pretrained%60%2C%20causing%20every%20RF-DETR%20parameterization%20to%20fail%20instead%20of%20running%20or%20skipping.%0A%0A---%0A%0AFor%20each%20issue%20above%2C%20determine%20whether%20it%20is%20valid%20and%20should%20be%20fixed.%20If%20so%2C%20fix%20it%20directly.&pr=682&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInCursorDark.svg?v=6"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInCursor.svg?v=6">Fix All in Cursor</picture></a> <a href="https://app.greptile.com/api/ide/codex?prompt=IMPORTANT%3A%20Work%20in%20the%20repository%20%22libreyolo%2Flibreyolo%22%20on%20the%20existing%20branch%20%22perf%2Fdetr-graphed-validation%22.%20Checkout%20that%20branch%20%E2%80%94%20do%20NOT%20create%20a%20new%20branch%20or%20open%20a%20new%20PR.%20Push%20your%20changes%20to%20%22perf%2Fdetr-graphed-validation%22.%0A%0A%23%23%23%20Issue%201%0Atests%2Funit%2Ftest_cuda_graph_detr_families.py%3A46%0A**RF-DETR%20unit%20tests%20require%20network%20access**%0A%0AIf%20%60transformers%60%20is%20installed%20but%20the%20DINOv2%20weights%20are%20not%20cached%20and%20Hugging%20Face%20is%20unavailable%2C%20this%20random-init%20construction%20still%20reaches%20%60AutoModel.from_pretrained%60%2C%20causing%20every%20RF-DETR%20parameterization%20to%20fail%20instead%20of%20running%20or%20skipping.%0A%0A---%0A%0AFor%20each%20issue%20above%2C%20determine%20whether%20it%20is%20valid%20and%20should%20be%20fixed.%20If%20so%2C%20fix%20it%20directly.&repo=libreyolo%2Flibreyolo&pr=682&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInCodexDark.svg?v=6"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInCodex.svg?v=6">Fix All in Codex</picture></a>

<sub>Reviews (2): Last reviewed commit: ["Defer the RF-DETR import so the gates co..."](https://github.com/libreyolo/libreyolo/commit/5a0581a9d7d3a07e79ca426a221c8519f2fcce33) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=49403975)</sub>

> Greptile also left **1 inline comment** on this PR.

### CODE REVIEW

File: tests/unit/test_cuda_graph_detr_families.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+from __future__ import annotations
+
+import pytest
+import torch
+
+from libreyolo.models.deimv2.model import LibreDEIMv2
+from libreyolo.models.dfine.model import LibreDFINE
+from libreyolo.models.ec.model import LibreEC
+from libreyolo.models.rfdetr.model import LibreRFDETR
```
[the remark this answers is not in the document]
Comment (EHxuban11) [reply tests/unit/test_cuda_graph_detr_families.py]: Valid catch, fixed in 5a0581a9.

`transformers` is only in the `rfdetr` extra, and RF-DETR's DINOv2 backbone imports it unconditionally, so the module-scope `LibreRFDETR` import failed collection for the whole file, including the four families (dfine, deimv2, ec, yolo9_e2e) that need no optional dependency at all.

Every model import now happens inside its case builder, which is the convention `test_cuda_graph_training.py` already uses for its RF-DETR parity case, and the RF-DETR builder calls `importorskip` on `transformers` (with `exc_type=ImportError`) so that family skips on its own.

Verified both ways: 20 passed with `transformers` installed, and 16 passed + 4 skipped with the import blocked via a meta-path hook (previously: collection error).
